lib/auth/auth.js

```diff
@@ -62,26 +62,23 @@ auth.check = (request, log, awsService, data) => {
     return { err: null, data: authInfo };
 };
 
-auth.doAuth = (request, log, cb, awsService, requestContext) => {
+auth.doAuth = (request, log, cb, awsService, requestContexts) => {
     const res = auth.check(request, log, awsService, request.query);
-    requestContext.setAuthType(res.authType);
-    requestContext.setSignatureVersion(res.signatureVersion);
-    requestContext.setSignatureAge(res.signatureAge);
+    requestContexts.forEach((requestContext) => {
+        requestContext.setAuthType(res.authType);
+        requestContext.setSignatureVersion(res.signatureVersion);
+        requestContext.setSignatureAge(res.signatureAge);
+    });
     if (res.err) {
         return cb(res.err);
     } else if (res.version === 2) {
         return vault.authenticateV2Request(res.data.accessKey,
             res.data.signatureFromRequest,
-            res.data.stringToSign, res.data.algo, requestContext, log,
-            (err, authInfo) => {
-                if (err) {
-                    return cb(err);
-                }
-                return cb(null, authInfo);
```

Comment:
Removing this means you will send back `undefined` instead of `null`, could you ensure there's no breakage anywhere due to this subtle change? (S3 tests come to mind).